### PR TITLE
overrides: add fixups for vllm's both wheel and source mode

### PR DESCRIPTION
**Pull request by Atry**
[Contribution](README.md#contributing) checklist (recommended but not always applicable/required):

- [x] There's an _[automated test](tests/default.nix)_ for this change
- [x] Commit messages or code include _references to related issues or PRs_ (including third parties)
- [x] Commit messages are _[conventional](https://www.conventionalcommits.org/)_ - examples from [the log](https://github.com/nix-community/poetry2nix/commits/master) include "feat: add changelog files to fixup hook", "fix(contourpy): allow wheel usage", and "test: add sqlalchemy2 test"

### CODE REVIEW

**Comment by Atry** [comment]
This PR partially fixes #1553

File: overrides/default.nix [hunk 1]
```diff
@@ -1366,7 +1378,7 @@ lib.composeManyExtensions [
           );
           llvm = pkgs."llvmPackages_${llvm_version}".llvm or (throw "LLVM${llvm_version} has been removed from nixpkgs; upgrade llvmlite or use older nixpkgs");
         in
-        {
+        lib.optionalAttrs (!old.src.isWheel or false) {
```
Comment (cpcloud) [inline overrides/default.nix:1381]:
```suggestion
        lib.optionalAttrs (!(old.src.isWheel or false)) {
```

File: overrides/default.nix [hunk 2]
```diff
@@ -1664,6 +1676,12 @@ lib.composeManyExtensions [
         }
       );
 
+      numba = super.numba.overridePythonAttrs (
+        old: {
+          propagatedBuildInputs = (old.propagatedBuildInputs or [ ]) ++ [ pkgs.tbb_2021_8 ];
```
Comment (cpcloud) [inline overrides/default.nix:1681]:
```suggestion
          propagatedBuildInputs = old.propagatedBuildInputs or [ ] ++ [ pkgs.tbb_2021_8 ];
```

Comment (cpcloud) [reply overrides/default.nix:1681]: Did tbb become a hard requirement for numba at some point?

File: overrides/default.nix [hunk 3]
```diff
@@ -3782,6 +3836,31 @@ lib.composeManyExtensions [
       pydantic = super.pydantic.overridePythonAttrs
         (old: { buildInputs = old.buildInputs or [ ] ++ [ pkgs.libxcrypt ]; });
 
+      vllm = super.vllm.overridePythonAttrs (old: {
+        autoPatchelfIgnoreMissingDeps = true;
+      } // (lib.optionalAttrs (!old.src.isWheel or false) rec {
```
Comment (cpcloud) [inline overrides/default.nix:3841]:
```suggestion
      } // (lib.optionalAttrs (!(old.src.isWheel or false)) rec {
```

**Comment by Atry** [comment]
@cpcloud The test failure seems not due to this PR

File: overrides/default.nix [hunk 4]
```diff
@@ -1705,6 +1717,12 @@ lib.composeManyExtensions [
         }
       );
 
+      numba = prev.numba.overridePythonAttrs (
+        old: {
+          propagatedBuildInputs = old.propagatedBuildInputs or [ ] ++ [ pkgs.tbb_2021_11 ];
```
Comment (cpcloud) [inline overrides/default.nix:1722]: Is the TBB dependency necessary?

Comment (Atry) [reply overrides/default.nix:1722]: Yes

Comment (cpcloud) [reply overrides/default.nix:1722]: I don't think TBB is required: https://numba.readthedocs.io/en/stable/user/threading-layer.html#which-threading-layers-are-available

There are multiple ways to back the threading layer.

If you want to TBB with numba with poetry2nix, it's up to the end user to add tbb as a dependency. We shouldn't opt people in to optional dependencies like this. Please revert the change.

Comment (cpcloud) [reply overrides/default.nix:1722]: It seems like the requirement for tbb is coming from vllm, so numba should be made to ignore missing autopatchelf deps and then you'll have to make sure in your application that uses vllm that you depend on tbb in some way.

Comment (cpcloud) [reply overrides/default.nix:1722]: Fixed.

**Review by cpcloud — APPROVED** [review]

**Review by Atry — COMMENTED** [review]
Looks good to me

**Comment by Atry** [comment]
vllm 0.3.3 works, but 0.4.1 crashes https://github.com/nix-community/poetry2nix/pull/1635

**Pull request merged**